### PR TITLE
Fix rate limiting tests

### PR DESCRIPTION
**Pull request by clokep**
These tests are from 1.19.1 (#8153) and seem to be broken after the merge from `release-v1.19.1` into `develop`.

### CODE REVIEW

File: tests/rest/client/v1/test_rooms.py [hunk 1]
```diff
@@ -688,7 +688,7 @@ class RoomJoinRatelimitTestCase(RoomBase):
     )
     def test_join_local_ratelimit(self):
         """Tests that local joins are actually rate-limited."""
-        for i in range(5):
+        for i in range(3):
```
Comment (clokep) [inline tests/rest/client/v1/test_rooms.py:691]: It is unclear to me whether this number is important or if it just eventually needs to be rate-limited.

**Comment by clokep** [comment]
@babolivier You might be best placed to review this?

File: tests/rest/client/v1/test_rooms.py (hunk 1 above)
Comment (babolivier) [reply tests/rest/client/v1/test_rooms.py:691]: This is the maximum number of requests you can send before the it gets rate-limited. I'm actually surprised it's passing, iirc that config lets slightly more than 3 requests through (5 as far as I could tell).

Comment (babolivier) [reply tests/rest/client/v1/test_rooms.py:691]: So we figured out where that was coming from (and how to fix it):

image

**Review by babolivier — APPROVED** [review]

**Pull request merged**